src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -42,13 +42,86 @@ namespace ErrorCodes
     extern const int DATABASE_ACCESS_DENIED;
     extern const int NOT_IMPLEMENTED;
     extern const int LOGICAL_ERROR;
+    extern const int BAD_ARGUMENTS;
 }
 
 namespace StorageObjectStorageSetting
 {
 extern const StorageObjectStorageSettingsBool allow_dynamic_metadata_for_data_lakes;
 }
 
+namespace
+{
+    void sanityCheckPartitioningConfiguration(
+        const ASTPtr & table_level_partition_by,
+        const ASTPtr & query_partition_by,
+        const StorageObjectStorage::ConfigurationPtr & configuration)
+    {
+        if (!table_level_partition_by && !query_partition_by)
+        {
+            // do we want to assert that `partitioning_style` is not set to something different style AND
+            // wildcard is not set either?
+            return;
+        }
+
+        if (table_level_partition_by && query_partition_by)
+        {
+            // should never happen because parser should not allow that
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Table level partition expression and query level partition expression can't be specified together, this is a bug");
+        }
+
+        static std::unordered_map<std::string, bool> partitioning_style_to_wildcard_acceptance =
+        {
+            {"auto", true},
+            {"hive", false}
+        };
+
+        if (!partitioning_style_to_wildcard_acceptance.contains(configuration->partitioning_style))
+        {
+            throw Exception(ErrorCodes::BAD_ARGUMENTS,
+                "Unknown partitioning style '{}'",
+                configuration->partitioning_style);
+        }
+
+        bool has_partition_wildcard = configuration->withPartitionWildcard();
+
+        if (has_partition_wildcard && !partitioning_style_to_wildcard_acceptance.at(configuration->partitioning_style))
```

Comment:
Please get that value once and save to a meaningfully named variable, because it is not entirely clear what does this bool flag conveys...

Reply:
You mean something like the below?

```
bool partitioning_style_supports_wildcard = partitioning_style_to_wildcard_acceptance.at(configuration->partitioning_style)

if (has_partition_wildcard && !partitioning_style_supports_wildcard)
{
...
}
```

Reply:
yes, that'll be nice